lib/service/awsoidc.go

```diff
@@ -128,6 +127,10 @@ func (cfg *AWSOIDCDeployServiceUpdaterConfig) CheckAndSetDefaults() error {
 		return trace.BadParameter("teleport cluster version required")
 	}
 
+	if cfg.UpgradeChannel == nil {
+		return trace.BadParameter("automatic upgrades channel required")
+	}
+
```

Comment:
Do we have guarantees that we're not breaking something here? Is there a legitimate non-AU case where we still want to run the AWS OIDC service?

Reply:
The updater requires a non-nil upgrade channel. If it were ever nil, then this would panic for example:

https://github.com/gravitational/teleport/blob/b58ad4aaf6c1a58ad39a81703a8faf77ce530ac4/lib/service/awsoidc.go#L188-L191

The channels come from proxy config. Proxy config channels are always non-nil, even if file config doesn't configure anything:
https://github.com/gravitational/teleport/blob/b58ad4aaf6c1a58ad39a81703a8faf77ce530ac4/lib/config/configuration.go#L1276-L1283

and in the default case where there is no file config, it's functionally equivalent to what we had before in the updater with `NewDefaultChannel`:

https://github.com/gravitational/teleport/blob/b58ad4aaf6c1a58ad39a81703a8faf77ce530ac4/lib/automaticupgrades/channel.go#L49-L52